internal/dcache/file_manager/file_manager.go

```diff
@@ -175,6 +177,22 @@ func (file *DcacheFile) SyncFile() error {
 		}
 		return true
 	})
+	common.Assert(err != nil)
+	return err
+}
+
+// Close and Finalize the file. writes are failed after this operation
+func (file *DcacheFile) CloseFile() error {
+	log.Debug("DistributedCache[FM]::CloseFilee : Close File for %s", file.FileMetadata.Filename)
+	err := file.SyncFile()
+	common.Assert(err == nil)
```

Comment:
sync must be called only when writing to file, so check for IsDcacheAllowWrites()
and add a comment why we are doing sync

// We stage application writes into StagedChunk and upload only when we have a full chunk.
// In case of last chunk being partial, we need to upload it now.